go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -39,11 +39,11 @@ func validateAndFinalizePod(
 	for index, container := range pod.Spec.Containers {
 		if container.Name == primaryContainerName {
 			hasPrimaryContainer = true
-			container.Resources = *flytek8s.ApplyResourceOverrides(ctx, container.Resources)
 			if taskCtx.TaskExecutionMetadata().GetOverrides() != nil && taskCtx.TaskExecutionMetadata().GetOverrides().GetResources() != nil {
 				resOverrides := taskCtx.TaskExecutionMetadata().GetOverrides().GetResources()
 				flytek8s.MergeResources(*resOverrides, &container.Resources)
 			}
+			container.Resources = *flytek8s.ApplyResourceOverrides(ctx, container.Resources)
```

Comment:
primary container resources should never be nil, but just in case should we check before dereferencing?

Reply:
just kidding, ApplyResourceOverrides handles initializing resources always

Reply:
Resolved offline.